hvantk/datasets/dataset_validator.py

```
@@ -0,0 +1,406 @@
+        if max_datasets:
+            datasets_to_validate = datasets_to_validate[:max_datasets]
+
+        logger.info(f"Starting batch validation of {len(datasets_to_validate)} UCSC datasets")
+
+        results = {}
+        for i, dataset in enumerate(datasets_to_validate, 1):
+            logger.info(f"Validating dataset {i}/{len(datasets_to_validate)}: {dataset.name}")
+            try:
+                result = self.validate_ucsc_dataset(dataset.name)
+                results[dataset.name] = result
+            except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)